### ISSUE TITLE
Compile options for Android?

### ISSUE THREAD

**Issue by gerroon** [koreader/koreader#1664]
Hi

I want to tackle compiling for Android and while doing that I want to play with different options etc. I do not see any doc or readme about that and it looks like there is no configure script either. So what are the compile time options that can be tested for Android?

thanks

**Comment by chrox** [comment]
Actually there is an ["Android" section](https://github.com/koreader/koreader#for-android-devices) in the readme file on instructions to build koreader for Android. The `Makefile` in `koreader` repo is just a wrapper of the heavy-lifting build script in the [`koreader-base`](https://github.com/koreader/koreader-base) repo and the [`android-luajit-launcher`](https://github.com/koreader/android-luajit-launcher) stuff especially for the launching things on Android.

**Comment by gerroon** [comment]
Thanks for the info.

What I am wondering if there are some config options that I can play with like memory layout, optimization etc? Messing with the whole build systemn is abit of out of my league.

**Comment by chrox** [comment]
I'm afraid you cannot config the memory layout in compile time, it's usually determined at running time, and running environment like the ROM version and what software you installed (such as Xposed) would affect it. And it's too early to talk about optimization. Let's first try to run it on those Android devices.

**Comment by gerroon** [comment]
"it's usually determined at running time:

That is interesting. The android version is incredibly flakey not just crashy. For example sometimes it just works, I mean I can open a book and browse. But most of the time it crashes with blanks or hangs there. It is a bit confusing because I cant determine when and how it decides to run.

Anyway I so want this app to work on Android without issues because it is such a wonderful application.

I have updated this bug with some logs and memory stuff if you have not seen it (which does not have any xposed or other trick stuff, right from Onyx)
https://github.com/koreader/koreader/issues/1643

Some of my tablets have xposed some do not and KR is as unstable as on each of them., I will try to provide more logs later today.

**Comment by chrox** [comment]
Yes, I already checked it out. But it will be more helpful if you report it like this:

1. the behavior of Koreader: crash or stuck
2. the maps file that captured in the mean time of your observation of 1
3. the logcat grepped with "luajit" in the mean time of your observation of 1

I wish I could debug myself but both of my Android devices run Koreader flawlessly so it's really a pity.

**Comment by gerroon** [comment]
The logcat is actually is done during the crash. If whatever you are looking for is not there that means that KR just disappears from memory which is how it looks to me when it crashes. 

In general KR just vapors. But the thing is that this happens at random times, sometimes when I click a menu item sometimes when I just flip a page.

Also this crashing is different on my Eink Android and normal android devices. On my C67ML itgenerally disappears and there is no pattern of crash. Sometimes I can click on the same menu item(generally the top menu or the file browsing menu) it just works, sometimes it just disappears.

I can generally read a book on my C67ML if I set my book properly and never touch a menu item ever again, however this is not so easy and it is just  a luck thing.

The behavior on tablets is different. For instances, they generally seem to have hard time starting. KR either never starts (blank screen) or starts but crashes very soon. Sometimes I can open a book and flip couple pages but other times it might not be able to open the same book that I was able to do couple days ago.


I also do not know what the map files are. If you tell me how to get them I will try to atatch them.
